optimuspy.py

```diff
@@ -134,8 +134,17 @@ def main(instance_name: str, view_name: str, process_name: str, executions: int,
     with TM1Service(**tm1_args) as tm1:
         original_performance_monitor_state = retrieve_performance_monitor_state(tm1)
         activate_performance_monitor(tm1)
+        all_tm1_cubes = filter(lambda c: not c.startswith("}"), tm1.cubes.get_all_names())
+
+        if cube_name is not None:
+            if cube_name in all_tm1_cubes:
+                model_cubes = [cube_name]
+            else:
+                logging.error(f"Provided cube '{cube_name}' does not exist")
+                exit(1)
```

Comment:
Maybe return `False` here instead or raise a `ValueError` and wrap the function call into a try-except block.

It's maybe just taste, but I prefer to reserve the use of exit to the "root" of the script and not use it in functions.